src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java

```diff
@@ -206,6 +206,18 @@ public static CryptoRandom getCryptoRandom(final Properties props)
                 } else {
                     throw initializerError;
                 }
+            } catch (final NoClassDefFoundError noClassDefFoundError) {
+                Throwable initializerError = noClassDefFoundError.getCause();
```

Comment:
There are some differences here between Java 8/11 and Java 17/21:
- Java 8/11: Here it will catch `NoClassDefFoundError`, but `NoClassDefFoundError.getCause` is `null`.
- Java 17/21: Here it will catch `NoClassDefFoundError`, and  `NoClassDefFoundError.getCause` is `ExceptionInInitializerError`, but `ExceptionInInitializerError.getCause` is null

So the check way here is a bit tricky, do you have any better suggestions? @garydgregory

Reply:
@LuciferYang 
Not off the top of my head...